### PR TITLE
Implement Print Operators for TimeStamp and Duration Classes

### LINKED ISSUE #765: Implement Print Operators For The Timestamp And Duration Classes
### Description of the task
See title. General idea is we should be able to write
````
Duration d  = Duration::fromSeconds(2);
std::cout << d;
```
and get something like 
```
2.00s
```


<!--
    What does this work depend on?
    What interface will this work use or create?
    What are the main components of the task?
    Where does this work fit in the larger project?

    It is important to define this task sufficiently so that an untrained
    team member can take it on and know where to start. Feel free to
    link to resources or other team member which could guide the assignee to
    complete the task
-->

### Acceptance criteria

<!--
    Checkbox list that outlines what needs to be done in order for this task
    to be considered "complete".

    Specify any implementation requirements such as data structures,
    functionalities, testing requirements, documentation, etc.
-->

- [ ] Print operators for the `Timestamp` class
- [ ] Print operators for the `Duration` class
- [ ] Item C

### Blocked By

<!--
    List all other issues that need to be completed before this one, ex:
    - #123
    - #374
-->

### PR DESCRIPTION
**Pull request by phelanm20**
### Description

Defines the << operator so that TimeStamp and Duration objects can be printed properly.

### Testing Done

    A single unit test was made for both classes. Test passes if printed output matches expected output.

### Resolved Issues

Resolves #765 

### Length Justification

 n/a 

### Review Checklist

**_It is the reviewers responsibility to also make sure every item here has been covered_**

- [x] **Function & Class comments**: All function definitions (usually in the `.h` file) should have a javadoc style comment at the start of them. For examples, see the functions defined in `thunderbots/software/geom`. Similarly, all classes should have an associated Javadoc comment explaining the purpose of the class.
- [x] **Remove all commented out code**
- [x] **Remove extra print statements**: for example, those just used for testing
- [x] **Resolve all TODO's**: All `TODO` (or similar) statements should either be completed or associated with a github issue
- [x] **Justify drops in code coverage**: If this PR results in a non-trivial drop in code coverage (a bot should post a coverage diagram as a comment), please justify why we can't test the code that's not covered.

### CODE REVIEW

File: src/software/util/time/duration.cpp [hunk 1]
```diff
@@ -59,3 +61,10 @@ Duration Duration::operator-(const Duration &duration) const
 {
     return Duration::fromSeconds(getSeconds() - duration.getSeconds());
 }
+
+std::ostream &operator<<(std::ostream &output_stream, const Duration &duration)
+{
+    output_stream << std::setprecision(2) << std::fixed << duration.getSeconds() << "s";
+
+    return output_stream;
+}
```
Comment (garethellis0) [inline src/software/util/time/duration.cpp]: nit; we're missing the newline at the end of this file (and github doesn't like that), just adding a empty line after this one should fix it.

File: src/software/util/time/duration_test.cpp [hunk 2]
```diff
@@ -133,3 +133,13 @@ TEST(DurationTest, test_subtraction_operator_with_negative_result)
...
+
+TEST(DurationTest, stream_operator)
+{
+    Duration d = Duration::fromSeconds(3.0);
+
+    std::stringstream out;
+    out << d;
+
+    EXPECT_EQ("3.00s", out.str());
+}
```
Comment (garethellis0) [inline src/software/util/time/duration_test.cpp]: missing newline at the end-of-file here, same suggestion applies as above

File: src/software/util/time/timestamp_test.cpp [hunk 3]
```diff
@@ -127,3 +128,13 @@ TEST(TimestampTest, test_subtraction_operator_with_timestamp_with_negative_resul
...
+
+TEST(TimestampTest, stream_operator)
+{
+    Timestamp t = Timestamp::fromSeconds(2.00);
+
+    std::stringstream out;
+    out << t;
+
+    EXPECT_EQ("2.00s", out.str());
+}
```
Comment (garethellis0) [inline src/software/util/time/timestamp_test.cpp]: missing newline at EOF here, same suggestion

**Review by garethellis0 — CHANGES_REQUESTED** [review]
Small changes here, looks good otherwise. Also we should delete the `src/thunderbots/CMakeLists.txt` file here, it's copy-over from the old build system.

**Comment by garethellis0** [comment]
Looks like there are some merge conflicts here, feel free to take a crack at resolving them yourself, or I'm happy to walk you through fixing them next weekend (since this PR isn't time-sensitive). This might be a good place to start: https://help.github.com/en/github/collaborating-with-issues-and-pull-requests/resolving-a-merge-conflict-using-the-command-line

**Comment by MathewMacDougall** [comment]
@phelanm20 any update on this?

**Review by kelvinkoon — DISMISSED** [review]
LGTM besides the newline comments Gareth pointed out already :+1:

**Review by MathewMacDougall — DISMISSED** [review]

File: src/software/util/time/duration.h [hunk 4]
```diff
@@ -1,5 +1,8 @@
 #pragma once
 
+
```
Comment (henryymliu) [inline src/software/util/time/duration.h]: uber nit; extra newline

File: src/software/util/time/timestamp_test.cpp [hunk 5]
```diff
@@ -4,6 +4,7 @@
 
 #include <stdexcept>
 
+
```
Comment (henryymliu) [inline src/software/util/time/timestamp_test.cpp]: uber nit: extra newline

File: src/software/util/time/timestamp.cpp [hunk 6]
```diff
@@ -70,3 +71,10 @@ Duration Timestamp::operator-(const Timestamp &timestamp) const
 {
     return Duration::fromSeconds(getSeconds() - timestamp.getSeconds());
 }
+
+std::ostream &operator<<(std::ostream &output_stream, const Timestamp &time)
+{
+    output_stream << std::setprecision(2) << std::fixed << time.getSeconds() << "s";
```
Comment (henryymliu) [inline src/software/util/time/timestamp.cpp:77]: Should `2` be a parameter/declared constant? @MathewMacDougall

**Review by garethellis0 — APPROVED** [review]
LGTM :+1: :tada: - Just as a note for the future, could you please leave a quick comment reply to each of my comments indicating what you changed? Just makes it a lot easier when I come back to re-review!

File: src/software/util/time/timestamp.cpp (hunk 6 above)
Comment (MathewMacDougall) [reply src/software/util/time/timestamp.cpp:77]: I don't really think so given the context and small usage scope.

**Review by MathewMacDougall — APPROVED** [review]
:tada:

**Review by henryymliu — APPROVED** [review]
🚢🤡

**Comment by MathewMacDougall** [comment]
@phelanm20 lets merge this before any more conflicts come up. We usually leave the final merge to the author, so just squash and merge this when you're ready.

**Pull request merged**